assets/src/scripts/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import SearchBarResults from './SearchBarResults.js';
+
+import 'whatwg-fetch';
+
+var strings = {
+	label:       'Search',
+	button:      'Submit',
+	placeholder: 'Search the site…',
+};
+
+export default class SearchBar extends React.Component {
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			query: this.props.query,
+			results: [],
+			focused: false,
+			loading: false,
+		};
+	}
+
+	render() {
+
+		var searchBarClasses = [ 'SearchBar' ];
+
+		if ( this.state.focused ) {
+			searchBarClasses.push( 'SearchBar-Focused' );
+			searchBarClasses.push( 'SearchBar-HasResults' );
+		}
+
+		if ( this.state.loading ) {
+			searchBarClasses.push( 'SearchBar-Loading' );
+		}
+
+		return <form className={ searchBarClasses.join(' ') } role="search" method="get" onSubmit={ (e) => e.preventDefault() }>
+			<label className="SearchBar_Label" htmlFor="site-search">{ strings.label }</label>
+			<div className="SearchBar_Container">
+				<input ref="input" className="SearchBar_Field" type="search" id="site-search" placeholder={ strings.placeholder } value={ this.state.query } onChange={ () => this.onSearch() } onFocus={ () => this.onFocus() } onBlur={ () => this.onBlur() } />
+				<button className="SearchBar_Submit" onClick={ () => this.onSearch }>
+					{ strings.button }
+					<span className={ this.state.loading ? 'Loading Loading-Active' : 'Loading' }></span>
+				</button>
+				<SearchBarResults results={ this.state.results } containerWidth={ this.props.containerEl.offsetWidth } />
+			</div>
+		</form>
+	}
+
+	onSearch( e ) {
+
+		var query = this.refs.input.value;
```

Comment:
you should be able access `event.target.value` to get the search query from the input field so you wouldn't need the `ref` attribute